### PR TITLE
Add spider for dacia_es

### PR DESCRIPTION
**Pull request by abbensid2**

### CODE REVIEW

**Comment by Cj-Malone** [comment]
@abbensid2 looks like there is a more efficient API: `https://www.dacia.es/wired/commerce/v1/dealers?page=454` I've not checked which countries it contains, but we could potentially merge a few of the spiders.

**Comment by abbensid2** [comment]
> @abbensid2 looks like there is a more efficient API: `https://www.dacia.es/wired/commerce/v1/dealers?page=454` I've not checked which countries it contains, but we could potentially merge a few of the spiders.

what do you think about this ?

<google-sheets-html-origin>

Country | Global | dacia_fr | dacia_es | dacia_gb | dacia_de
-- | -- | -- | -- | -- | --
AL | 1 |   |   |   |  
AT | 168 |   |   |   |  
BA | 10 |   |   |   |  
BE | 209 |   |   |   |  
BG | 21 |   |   |   |  
CH | 216 |   |   |   |  
CZ | 62 |   |   |   |  
DE | 937 |   |   |   | 980
DK | 65 |   |   |   |  
DZ | 58 |   |   |   |  
EE | 11 |   |   |   |  
ES | 531 |   | 381 |   |  
FI | 3 |   |   |   |  
FR | 2841 |   | 3369 |   |  
GB | 201 |   |   | 201 |  
HR | 36 |   |   |   |  
HU | 50 |   |   |   |  
IE | 34 |   |   |   |  
IT | 726 |   |   |   |  
LT | 8 |   |   |   |  
LU | 13 |   |   |   |  
LV | 7 |   |   |   |  
MA | 51 |   |   |   |  
ME | 2 |   |   |   |  
MK | 2 |   |   |   |  
NL | 156 |   |   |   |  
NO | 1 |   |   |   |  
PL | 99 |   |   |   |  
PT | 98 |   |   |   |  
RO | 142 |   |   |   |  
RS | 17 |   |   |   |  
RU | 153 |   |   |   |  
SE | 147 |   |   |   |  
SI | 45 |   |   |   |  
SK | 29 |   |   |   |  
TR | 127 |   |   |   |  
UA | 39 |   |   |   |  
XK | 1 |   |   |   |  
Totoal | 7317 |   |   |   |

**Comment by abbensid2** [comment]
It has latitude and longitude coordinates

<google-sheets-html-origin>

Country | Global | dacia_fr | dacia_es | dacia_gb | dacia_de
-- | -- | -- | -- | -- | --
DE | 809 |   |   |   | 821
ES | 529 |   | 381 |   |  
FR | 2785 | 3313 |   |   |  
GB | 199 |   |   | 199 |

File: locations/spiders/dacia_es.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        for data in response.json():
+            item = GeojsonPointItem()
+            item["ref"] = data.get("birId")
+            item["name"] = data.get("name")
+            item["name"] = data.get("name")
+            item["country"] = data.get("country")
+            item["lat"] = data.get("geolocalization", {}).get("lat")
+            item["lon"] = data.get("geolocalization", {}).get("lon")
+            item["city"] = data.get("regionalDirectorate")
+            item["city"] = data.get("regionalDirectorate")
```
Comment (jancallewaert) [inline locations/spiders/dacia_es.py:32]:
```suggestion
            item["city"] = data.get("regionalDirectorate")
```

**Comment by Cj-Malone** [comment]
> Country | Global | dacia_fr
> -- | -- | --
> FR | 2841 | 3369

This looks a little weird to me, I've still not looked at the data, but I think there may be 2 possibilities. I think the most likely, `dacia_fr` is also picking up some POIs in surrounding countries, if so, I'd vote for the global API. Or, `dacia_fr` API actually has more FR POIs, if that's the case I'd vote for 2 spiders, `dacia_fr` and a global `dacia` for everywhere else.

**Comment by abbensid2** [comment]
> > Country
> > Global
> > dacia_fr
> > 
> > 
> > 
> > 
> > FR
> > 2841
> > 3369
> 
> This looks a little weird to me, I've still not looked at the data, but I think there may be 2 possibilities. I think the most likely, `dacia_fr` is also picking up some POIs in surrounding countries, if so, I'd vote for the global API. Or, `dacia_fr` API actually has more FR POIs, if that's the case I'd vote for 2 spiders, `dacia_fr` and a global `dacia` for everywhere else.

the output of spider dacia_fr only contains the POI FR.
I run the global spider on different IPs, and it turned out different:
# IP Spain: 7394
# IP France: 7381
# IP United States: 6677
# IP United Kingdom: 7382

**Comment by abbensid2** [comment]
> `dacia_fr` API actually has more FR POIs, if that's the case I'd vote for 2 spiders, `dacia_fr` and a global `dacia` for everywhere else.

the output of spider `dacia_fr` only contains the POIs FR.
I run the `global spider` on different IPs, and it turned out different:
IP Spain: 7394
IP France: 7381
IP United States: 6677
IP United Kingdom: 7382

**Comment by Cj-Malone** [comment]
That is weird. I think I'd still vote for paging the `dealers` endpoint over a lat/lon search. It'll be more efficient and predictable. And it's just politer for the API provider to not hammer away.

> different IPs, different results

We have the theoretical `requires_proxy = True`, it's not used yet but putting it in some spiders for when we do.

@iandees I think `requires_proxy = True` should mean run it in the country code in the spider name eg "jdsports_gb" gets run via a GB ip. It could default to US when there isn't a country code in the spider name. But I think we could then expand it to `requires_proxy = "ES"` overriding the spider name or for applying it to global spiders like this.

**Comment by Cj-Malone** [comment]
Just had another quick look at the API, I can't reproduce different `totalElements` from a GB or US ip, I always get `9078`. I've not crawled through the pages though, so I've that's not accounting for `active` or anything else. It also looks like we can filter by country: `https://www.dacia.es/wired/commerce/v1/dealers?country=FR` and the TLD doesn't seem to alter the data, eg: `https://www.dacia.fr/wired/commerce/v1/dealers?country=FR` also gives `"totalElements": 3558` to me.

**Comment by abbensid2** [comment]
There are duplicate items.
[scrapy.log](https://github.com/alltheplaces/alltheplaces/files/10344522/scrapy.log)

**Comment by abbensid2** [comment]
PR is closed, merged to PR: #4589

**Pull request closed without merging**